src/cpp/modules/manualEigenVector/ManualEigenVectorGMM.cpp

```diff
@@ -0,0 +1,78 @@
+// ManualEigenVectorGMM.cpp : Defines the exported functions for the DLL.
+#include "ManualEigenVectorGMM.h"
+#include "../../shared/gmm.h"
+#include "gmm_vector_d.h"
+#include "memory_size.h"
+
+#include <iostream>
+#include <iomanip>
+#include <memory>
+
+// This function must be called before any other function.
+void ManualEigenVectorGMM::prepare(GMMInput&& input)
+{
+    _input = input;
+    int Jcols = (_input.k * (_input.d + 1) * (_input.d + 2)) / 2;
+    _output = { 0,  std::vector<double>(Jcols) };
+
+    // If it is not enough memory in system to store all data that
+    // manualEigenVector need, then executing will interrupt
+    // to prevent system deadlock
+    unsigned long long int memory_size = get_memory_size();
+    // If memory_size != 0 then check memory size,
+    // else OS is undefined by get_memory_size()
+    // and we try to execute module
+    if (memory_size != 0) {
+        unsigned long long int d = input.d;
+        unsigned long long int k = input.k;
+        unsigned long long int n = input.n;
+        unsigned long long int icf_sz = d * (d + 1) / 2;
+        unsigned long long int need_memory =
+            (
+            (k + k * d + k * icf_sz) // J
+            + k + (d * k) + (d * n)  // eigen wrappers
+            + k + (k * d * d) // sum_qs, Qs
+            + (d * n) + (d * n) + (k * n) //xcentered, Qxcentered, main_term
+            + (k * d * d) + (k * d * d) // tmp_means_d, tmp_qs_d
+            + (k * (icf_sz - d) * n) // tmp_L_d
+            + n + k // mX, semX
+            ) * sizeof(double);
+        // We would like to prevent system even from freezes
+        // so we don't run module if it's less than 2GB RAM
+        // left for normal operation of the system
+        need_memory += 2UL * 1024UL * 1024UL * 1024UL; // + 2GB
```

Comment:
// When the system doesn't have enough memory, running the test is not only pointless, because the results will be spoiled by excessive swapping, but may also be harmful to the functioning of the OS. Too much memory pressure may (and does in practice) render the system unresponsive and lead to the invocation of OOM Killer (or its equivalent).